pkg/storaged/btrfs/subvolume.jsx

```diff
@@ -0,0 +1,106 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+import React from "react";
+
+import { CardBody } from "@patternfly/react-core/dist/esm/components/Card/index.js";
+import { DescriptionList } from "@patternfly/react-core/dist/esm/components/DescriptionList/index.js";
+
+import { StorageCard, StorageDescription, new_card, new_page } from "../pages.jsx";
+import { StorageUsageBar } from "../storage-controls.jsx";
+import { get_fstab_config_with_client } from "../utils.js";
+import { btrfs_usage } from "./utils.jsx";
+import { mounting_dialog } from "../filesystem/mounting-dialog.jsx";
+import { check_mismounted_fsys, MismountAlert } from "../filesystem/mismounting.jsx";
+import { is_mounted, mount_point_text, MountPoint } from "../filesystem/utils.jsx";
+import client from "../client.js";
+
+const _ = cockpit.gettext;
+
+function subvolume_unmount(volume, subvol, mount_point) {
+    const block = client.blocks[volume.path];
+    // HACK: generalise this? Instead of a global findmnt state, put the active mount point in subvol.
+    subvol.mount_point = mount_point;
+    mounting_dialog(client, block, "unmount", null, subvol);
+}
+
+function subvolume_mount(volume, subvol) {
+    const block = client.blocks[volume.path];
+    mounting_dialog(client, block, "mount", null, subvol);
+}
+
+export function make_btrfs_subvolume_page(parent, volume, subvol) {
+    const actions = [];
+
+    const use = btrfs_usage(client, volume);
+    const backing_block = client.blocks[volume.path];
+    const content_block = client.blocks_fsys[volume.path];
```

Comment:
This is confusing... `client.blocks´ is a o.fd.UDisks2.Block interface and `client.blocks_fsys` is a `o.fd.UDisks2.Filesystem` interface for the the same object.

The rest of the code uses backing_block and content_block if there might be two blocks invovled for the same page, such as with encryption. Then backing_block is the block with the LUKS signature on it, and content_block is the device mapper block with the filesystem signature.

I don't think this applies for btrfs at all, and you are not really using content_block below anyway, except for check_mismounted_fsys. "Luckily", that function only uses content_block.path, which makes it correct, but only accidentally.

So just have "block" here, and call `check_mismounted_fsys(block, block, ...)`.